lib/rubocop/cop/style/numbered_parameters.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Use numbered parameters only for single-line block.
+      #
+      # @example EnforcedStyle: allow_single_line (default)
+      #   # bad
+      #   collection.each do
+      #     puts _1
+      #   end
+      #
+      #   # good
+      #   collection.each { puts _1 }
+      #
+      # @example EnforcedStyle: disallow
+      #   # bad
+      #   collection.each { puts _1 }
+      #
+      #   # good
+      #   collection.each { |item| puts item }
+      #
+      class NumberedParameters < Base
+        include ConfigurableEnforcedStyle
+        extend TargetRubyVersion
+
+        MSG = 'Avoid using numbered parameters.'
+        MSG_MULTI_LINE = 'Avoid using numbered parameters for multi-line blocks.'
+
+        minimum_target_ruby_version 2.7
+
+        def on_numblock(node)
+          return unless style == :disallow || node.multiline?
+
+          add_offense(node, message: style == :disallow ? MSG : MSG_MULTI_LINE)
```

Comment:
I tend to avoid `unless` with boolean logic, how about this?

```suggestion
          if style == :disallow
            add_offense(node, message: MSG)
          elsif node.multiline?
            add_offense(node, message: MSG_MULTILINE)
          end
```

Reply:
Great suggestion, amended